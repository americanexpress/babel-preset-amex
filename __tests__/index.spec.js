/*
 * Copyright (c) 2017 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */

const preset = require('..');

describe('babel-preset-amex', () => {
  it('exports a function', () => {
    expect(preset).toEqual(expect.any(Function));
    expect(preset()).toEqual(expect.any(Object));
  });

  it('includes setting for modern browsers', () => {
    expect(preset().env.modern).toEqual(expect.any(Object));
  });

  it('includes an array of presets', () => {
    expect(preset().presets).toEqual(expect.any(Array));
    expect(preset().presets.length).toBe(2);
    preset().presets.forEach(p => expect(p).toEqual(expect.any(Object)));
  });

  it('includes an array of plugins', () => {
    expect(preset().plugins).toEqual(expect.any(Array));
    expect(preset().plugins.length).toBe(5);
    preset().plugins.forEach((plugin) => {
      // It should be either a function
      try {
        expect(plugin).toEqual(expect.any(Function));
        return;
      } catch (e) { /* do nothing */ }

      // or an array containing a function and an object
      expect(plugin).toEqual(expect.any(Array));
      expect(plugin.length).toBe(2);
      expect(plugin[0]).toEqual(expect.any(Function));
      expect(plugin[1]).toEqual(expect.any(Object));
    });
  });
});
