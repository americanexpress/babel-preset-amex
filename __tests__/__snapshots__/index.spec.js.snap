// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-preset-amex allows options to be passed to plugins 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-syntax-dynamic-import",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-export-default-from",
    "@babel/plugin-proposal-optional-chaining",
    "babel-plugin-transform-react-remove-prop-types",
  ],
  "presets": Array [
    Array [
      Object {
        "default": "@babel/preset-env",
      },
      Object {
        "exclude": Array [
          "@babel/plugin-transform-regenerator",
        ],
        "targets": Object {
          "browsers": Array [
            "last 2 versions",
            "not dead",
          ],
          "node": "current",
        },
      },
    ],
    Array [
      Object {
        "default": "@babel/preset-react",
      },
      Object {},
    ],
  ],
}
`;

exports[`babel-preset-amex returns modern preset for env and option 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-syntax-dynamic-import",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-export-default-from",
    "@babel/plugin-proposal-optional-chaining",
    "babel-plugin-transform-react-remove-prop-types",
  ],
  "presets": Array [
    Array [
      Object {
        "default": "@babel/preset-env",
      },
      Object {
        "targets": Object {
          "browsers": Array [
            "last 2 Chrome versions",
            "last 2 ChromeAndroid versions",
            "last 2 Edge versions",
            "last 2 Firefox versions",
            "last 2 iOS versions",
            "last 2 Safari versions",
          ],
          "node": "current",
        },
      },
    ],
    Array [
      Object {
        "default": "@babel/preset-react",
      },
      Object {},
    ],
  ],
}
`;

exports[`babel-preset-amex returns modern preset for env and option 2`] = `
Object {
  "plugins": Array [
    "@babel/plugin-syntax-dynamic-import",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-export-default-from",
    "@babel/plugin-proposal-optional-chaining",
    "babel-plugin-transform-react-remove-prop-types",
  ],
  "presets": Array [
    Array [
      Object {
        "default": "@babel/preset-env",
      },
      Object {
        "targets": Object {
          "browsers": Array [
            "last 2 Chrome versions",
            "last 2 ChromeAndroid versions",
            "last 2 Edge versions",
            "last 2 Firefox versions",
            "last 2 iOS versions",
            "last 2 Safari versions",
          ],
          "node": "current",
        },
      },
    ],
    Array [
      Object {
        "default": "@babel/preset-react",
      },
      Object {},
    ],
  ],
}
`;

exports[`babel-preset-amex returns server only config when given serverOnly option 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-syntax-dynamic-import",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-export-default-from",
    "@babel/plugin-proposal-optional-chaining",
    "babel-plugin-transform-react-remove-prop-types",
  ],
  "presets": Array [
    Array [
      Object {
        "default": "@babel/preset-env",
      },
      Object {
        "targets": Object {
          "node": "current",
        },
      },
    ],
    Array [
      Object {
        "default": "@babel/preset-react",
      },
      Object {},
    ],
  ],
}
`;

exports[`babel-preset-amex returns server only config when given serverOnly option 2`] = `
Object {
  "plugins": Array [
    "@babel/plugin-syntax-dynamic-import",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-export-default-from",
    "@babel/plugin-proposal-optional-chaining",
    "babel-plugin-transform-react-remove-prop-types",
  ],
  "presets": Array [
    Array [
      Object {
        "default": "@babel/preset-env",
      },
      Object {
        "targets": Object {
          "node": "current",
        },
      },
    ],
    Array [
      Object {
        "default": "@babel/preset-react",
      },
      Object {},
    ],
  ],
}
`;
